Sofa/Component/Engine/Transform/src/sofa/component/engine/transform/DilateEngine.inl

```diff
@@ -92,13 +81,21 @@ void DilateEngine<DataTypes>::doUpdate()
     const Real minThickness = d_minThickness.getValue();
 
     WriteOnlyAccessor<Data<VecCoord> > out = d_outputX;
+    WriteOnlyAccessor<Data<VecCoord> > normals = d_normals;
 
     const int nbp = in.size();
     const int nbt = triangles.size();
     const int nbq = quads.size();
 
-    WriteOnlyAccessor<Data<VecCoord> > normals = d_normals;
+    if(nbt == 0)
+        msg_warning() << "No input mesh (no triangle)";
+    if(nbq == 0)
+        msg_warning() << "No input mesh (no quad)";
+    if(nbp == 0)
```

Comment:
I know this is not the PR scope but if nbp is null, the component can't work.
I would expect a componentState to invalid and return.
Espacially if the list of triangles/quads is not empty and the list of point is empty the loop behind in the code will create a segfault.